salt/modules/zabbix.py

```diff
@@ -2426,7 +2426,7 @@ def run_query(method, params, **kwargs):
             ret = _query(method, params, conn_args['url'], conn_args['auth'])
             if isinstance(ret['result'], bool):
                 return ret['result']
-            return ret['result'] if ret['result'] else False
+            return ret['result'] if (ret['result'] == True or len(ret['result']) > 0) else False
```

Comment:
To be lint proper this should probably read something like:
`return ret['result'] if (bool(ret['result']) or len(ret['result']) > 0) else False`